### DISCUSSION TITLE
Validate client certificate ourselves

### DISCUSSION THREAD

**Discussion by drauch** [dotnet/yarp#774] [Q&A] [answered]
Is it already possible to perform client certificate validation ourselves? We want to build a proxy that performs extensive and in-depth client certificate validation before passing on the request.

**Comment by Tratcher** [comment, marked as answer]
Yes, see https://docs.microsoft.com/en-us/aspnet/core/security/authentication/certauth?view=aspnetcore-5.0.
https://microsoft.github.io/reverse-proxy/articles/authn-authz.html
https://microsoft.github.io/reverse-proxy/articles/transforms.html#clientcert
